src/components/Popover/Popover.tsx

```diff
@@ -61,29 +61,33 @@ export interface PopoverProps {
    * The placement (position) of the Popover relative to its trigger.
    */
   placement?: Placement;
-  /**
-   * The target element where the Popover will be portaled to, when `withPortal === true`.
-   */
-  portalTarget?: HTMLElement;
   /**
    * Whether you want to trap focus in the Popover element when it is open.
    * Read more about focus traps:
    * [Here](https://allyjs.io/tutorials/accessible-dialog.html#trapping-focus-inside-the-dialog)
    */
   trapFocus?: boolean;
+  /**
+   * Additional props to be spread to rendered element
+   */
+  [x: string]: any; // eslint-disable-line
+} & ({
   /**
    * Whether the element should be rendered outside its DOM structure
    * for reasons of placement. Use this when the element is being cut-off or
    * re-positioned due to lack of space in the parent container.
    * NOTE: use `portalTarget` to render the element onto a custom container,
    * otherwise it will be rendered to the `body` element by default.
    */
-  withPortal?: boolean;
-  /**
-   * Additional props to be spread to rendered element
+  withPortal: true;
+   /**
+   * The target element where the Popover will be portaled to, when `withPortal === true`.
    */
-  [x: string]: any; // eslint-disable-line
-}
+  portalTarget?: HTMLElement;
```

Comment:
In this case, do we still need to mark this as optional with the `?` feels like it should be required if `withPortal` is true.

Reply:
Having it default to document.body seemed nice! Having that default was the only thing keeping me from making portalTarget required, so if you want to get rid of that default we can make this required!

Reply:
Juan and I discussed this over Slack, we're moving forward with the change to require portalTarget if withPortal is true